include/CL/sycl/buffer.hpp

```diff
@@ -484,10 +491,33 @@ class buffer
       way to write back some data or with some data sharing with the
       host that can not be undone
   */
+  void set_final_data(shared_ptr_class<T> finalData) {
+    implementation->implementation->set_final_data(std::move(finalData));
+  }
+
+
   void set_final_data(weak_ptr_class<T> finalData) {
     implementation->implementation->set_final_data(std::move(finalData));
   }
 
+
+  /** Disable write-back on buffer destruction.
+   */
+  void set_final_data(std::nullptr_t) {
+    implementation->implementation->set_final_data(nullptr);
+  }
+
+
+  /** WARNING: the user has to ensure that the object refered to by the
```

Comment:
This is not a comment describing the purpose of this function. Add it in first position.